src/createXcmTypes/SystemToSystem.ts

```diff
@@ -0,0 +1,302 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+
+import type { ApiPromise } from '@polkadot/api';
+import { u32 } from '@polkadot/types';
+import type {
+	MultiAssetsV2,
+	VersionedMultiAssets,
+	VersionedMultiLocation,
+	WeightLimitV2,
+} from '@polkadot/types/interfaces';
+import type { XcmV3MultiassetMultiAssets } from '@polkadot/types/lookup';
+import { isEthereumAddress } from '@polkadot/util-crypto';
+
+import { SYSTEM_PARACHAINS_IDS } from '../consts';
+import { getChainIdBySpecName } from '../createXcmTypes/util/getChainIdBySpecName';
+import { BaseError } from '../errors';
+import type { Registry } from '../registry';
+import { getFeeAssetItemIndex } from '../util/getFeeAssetItemIndex';
+import { normalizeArrToStr } from '../util/normalizeArrToStr';
+import { MultiAsset, MultiAssetInterior } from './../types';
+import {
+	CreateAssetsOpts,
+	CreateFeeAssetItemOpts,
+	ICreateXcmType,
+	IWeightLimit,
+} from './types';
+import { dedupeMultiAssets } from './util/dedupeMultiAssets';
+import { fetchPalletInstanceId } from './util/fetchPalletInstanceId';
+import { getSystemChainTokenSymbolGeneralIndex } from './util/getTokenSymbolGeneralIndex';
+import { isRelayNativeAsset } from './util/isRelayNativeAsset';
+import { sortMultiAssetsAscending } from './util/sortMultiAssetsAscending';
+
+export const SystemToSystem: ICreateXcmType = {
+	/**
+	 * Create a XcmVersionedMultiLocation type for a beneficiary.
+	 *
+	 * @param api ApiPromise
+	 * @param accountId The accountId of the beneficiary
+	 * @param xcmVersion The accepted xcm version
+	 */
+	createBeneficiary: (
+		api: ApiPromise,
+		accountId: string,
+		xcmVersion?: number
+	): VersionedMultiLocation => {
+		if (xcmVersion == 2) {
+			const X1 = isEthereumAddress(accountId)
```

Comment:
I dont think we need the `isEthereum` here, and we can get rid of `AccountId20` since the system parachain probably wont support ethereum addresses.